std/path.d

```diff
@@ -3883,11 +3886,16 @@ string expandTilde(string inputPath) nothrow
                         break;
                     }
 
-                    if (errno != ERANGE)
+                    if (errno != ERANGE &&
+                        // On FreeBSD and OSX, errno can be left at 0 instead of set to ERANGE
+                        errno != 0)
                         onOutOfMemoryError();
 
                     // extra_memory isn't large enough
-                    extra_memory_size *= 2;
+                    import core.checkedint : mulu;
+                    bool overflow;
+                    extra_memory_size = mulu(extra_memory_size, 2, overflow);
+                    if (overflow) assert(0);
```

Comment:
The message is printed if not built with -release, because in that case, `assert(0)` is compiled in like any assertion would be and throws an `AssertError`. However, with -release, when assertions are compiled out, it becomes a HLT instruction, and there is no message.